src/analysis/processing/qgsalgorithmextractbylocation.h

```diff
@@ -55,8 +55,16 @@ class QgsLocationBasedAlgorithm : public QgsProcessingAlgorithm
 
   private:
 
+    friend class QgsSelectByLocationAlgorithm;
+    friend class QgsExtractByLocationAlgorithm;
```

Comment:
Is there a reason for friend instead of protected variables?

Reply:
@troopa81 , not really no.

Reply:
do you mind changing it?

Reply:
@troopa81 , done.